content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +323,60 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+A NetworkPolicy is applied to a selected pod, when isolation rules are enforced
+for that pod. At that point pod can never establish an outbound connection or receive
+an inbound connection that is not allowed by the NetworkPolicy. Allow rules will be applied
+eventually after the isolation rules (that includes applying both at the same time).
+That means, that once the NetworkPolicy is applied to a pod, the pod should be able to handle situations,
+when it can't establish a connection that is allowed by an existing NetworkPolicy for some time.
+
+Every NetworkPolicy will be applied to all selected pods eventually, but there is no
+way to tell from the existing Kubernetes API when exactly the network policy is applied to a specific
+pod. The only way to check if the network policy is applied to a pod, is to run a connectivity check.
+
+Network plugins may implement extra `status.conditions` for pod readiness based on the
+status of network policies. These conditions then may be used by 
+[Pod readiness gate](/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate).
+
+## NetworkPolicy and `hostNetwork` pods
+
+NetworkPolicy behaviour for `hostNetwork` pods is undefined, when
```

Comment:
"undefined" makes it sound like this is an unanswered question, but AFAIK literally every network plugin implements this the same way: a `podSelector` (whether spec or gress) never selects `hostNetwork` pods.

(In theory, Cilium (or any other plugin using at least some eBPF) could do something better here, by applying some rules based on cgroups rather than only using IP addresses. But AFAIK nobody actually currently does this, so it seems like we should just officially bless the current behavior and revisit this in NPv2.)

Reply:
I may know one plugin that interprets peer hostNetwork pods as nodeIP, since it is the only way (I can think of) to allow kubeapi pods without hardcoding master node ips.
I am not sure if we can say that hostNetwork pods from the `spec` are always ignored, just because we don't know of any plugins that do something else. Maybe we better say something like "most network plugins" will ignore, but it is still officially undefined?